purchase_order_line_schedule/init_hooks.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2021 ForgeFlow, S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html)
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def pre_create_schedule_lines(cr):
+    _logger.info(
+        "Pre-creating purchase.order.line.schedule records for "
+        "existing purchase orders"
+    )
+    cr.execute(
+        """
+        INSERT INTO purchase_order_line_schedule
+            (create_uid, create_date, write_uid, write_date,
+             order_id, order_line_id, date_planned, product_qty,
+             qty_received_manual, product_id, company_id)
+        SELECT create_uid, create_date, write_uid, write_date,
+               order_id, id, date_planned, product_qty,
+               qty_received_manual, product_id, company_id
+        FROM purchase_order_line
+    """
+    )
```

Comment:
`ERROR: null value in column "date_planned" violates not-null constraint `

Seems like date_planned can be null on POL, I suggest:

        ```
INSERT INTO purchase_order_line_schedule
            (create_uid, create_date, write_uid, write_date,
             order_id, order_line_id, date_planned, product_qty,
             qty_received_manual, product_id, company_id)
        SELECT pol.create_uid, pol.create_date, pol.write_uid, pol.write_date,
               pol.order_id, pol.id, coalesce(pol.date_planned, po.date_planned, po.date_order) , pol.product_qty,
               pol.qty_received_manual, pol.product_id, pol.company_id
        FROM purchase_order_line AS pol, purchase_order AS po
        WHERE pol.order_id = po.id
```

Reply:
@JordiBForgeFlow 
Why didn't you process this feedback?

Reply:
@CasVissers-360ERP I missed it. Will review.

Reply:
Feedback still valid in my opinion. Still got the error.